Library/Monitoring/Service.php

```diff
@@ -0,0 +1,125 @@
+<?php
+namespace Vpg\Disturb\Monitoring;
+
+use \Phalcon\Config;
+use \Phalcon\Mvc\User\Component;
+use \Elasticsearch;
+
+use Vpg\Disturb\Workflow;
+use Vpg\Disturb\Core;
+use Vpg\Disturb\Core\Storage;
+
+/**
+ * Class WorkerService monitoring
+ *
+ * @package  Disturb\Monitoring
+ * @author   Jérôme Bourgeais <jbourgeais@voyageprive.com>
+ * @license  https://github.com/vpg/disturb/blob/master/LICENSE MIT Licence
+ */
+class Service extends Component
+{
+
+    /**
+     * ContextStorage constructor
+     *
+     * @param Json $config config
+     *
+     * @throws ContextStorageException
+     */
+    public function __construct(Config $config)
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $this->config = new Workflow\WorkflowConfigDto($config);
+        $this->initClient();
+    }
+
+    /**
+     * Initialization of Elasticsearch Client
+     *
+     * @throws ContextStorageException
+     *
+     * @return void
+     */
+    private function initClient()
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $storageHost = $this->config->getStorageHost();
+        $this->storageClient = Storage\StorageAdapterFactory::get(
+            $this->config,
+            Storage\StorageAdapterFactory::USAGE_MONITORING
+        );
+    }
+
+    /**
+     * Registers a worker into the monitoring sys
+     *
+     * @param string $workerCode the worker's code to register
+     *
+     * @return void
+     */
+    public function workerBeat(string $workerCode)
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $workerHash = [
+            'heartBeatAt' => Date('Y-m-d H:i:s')
+        ];
+        $this->storageClient->save($workerCode, $workerHash);
+    }
+
+    /**
+     * Registers a worker into the monitoring sys
+     *
+     * @param string $workerCode the worker's code to register
+     * @param int    $pid        the worker's pid
+     *
+     * @return void
+     */
+    public function workerStarted(string $workerCode, int $pid)
```

Comment:
Change order in function name startWorker

Reply:
infact, here we monitor a past action "worker started" 
if we define a "startWorker" func it's supposed to "start a worker"
Maybe you want the func to start w/ a verb. why not `logWorkerStarted` ?